src/main/java/com/notably/logic/suggestion/handler/SuggestionArgHandler.java

```diff
@@ -0,0 +1,20 @@
+package com.notably.logic.suggestion.handler;
+
+import java.util.Optional;
+
+import com.notably.logic.suggestion.generator.SuggestionGenerator;
+
+/**
+ * Represents a Suggestion Argument Handler that is able to handle user input into a SuggestionGenerator.
+ *
+ * @param <T> Generic type of SuggestionGenerator.
+ */
+public interface SuggestionArgHandler<T extends SuggestionGenerator> {
+    /**
+     * Handles user input into a suggestion generator and returns it.
+     *
+     * @param userInput The user input.
```

Comment:
Is it more descriptive to say `The arguments part of the user input`, then explains a bit on what you mean by the `arguments` part in the interface description?

Reply:
Ok!